README.md

```diff
@@ -1,65 +1,100 @@
 
-## Submodules
-This repository uses git submodules. Ensure to keep them in sync after clone with
+# Installation
+
+## Repo
+
+This repository uses [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules). To checkout all the necessary code begin with the following command:
 ```console
-> git submodule update --init
+> git clone --recurse-submodules https://github.com/unhuman-io/obot-controller.git
+```
+To keep the submodules tidy it is encouraged to setup your new git repo with the following options:
+```bash
+git config --global submodule.recurse true
+git config --global pull.ff only
 ```
-and when pulling or checking out
+Finally, checkout the latest branch and make sure it is up-to-date:
+```console
+> git checkout --recurse-submodules develop
+```
+
+## Build
+The firmware is designed around a **gcc** toolchain for ARM and the compiler (`arm-none-eabi`) can be manually downloaded from the [ARM website](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads) or automatically with a script.
+
+Prepare the environment by installing the following modules:
+```console
+> sudo apt-get install curl wget xz-utils dfu-util make
+```
+Then run the install script from within the root `obot-controller/` directory, which will install **gcc** into `obot-controller/motorlib/gcc/`:
+```console
+> ./motorlib/scripts/install_gcc.sh
+```
+
+## Debugging
+The firmware is designed to communicate via USB, which allows interactive debugging and tuning via a command-line utility `motor_util`. Installation of the utility can be achieved by:
+
+```bash
+sudo apt install -y linux-headers-$(uname -r)
+curl https://raw.githubusercontent.com/unhuman-io/obot/main/install-obot.sh > install-obot.sh
+chmod +x install-obot.sh
+./install-obot.sh
+```
+
+This will install a realtime USB kernel module for communicating with the motor controller and install a set of utilities for working with the device. The utilities can be upgraded with `update-obot`.
+
+# Development
+## Repo
+This repository uses [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules). It is useful to periodically sync the various repos after the initial `clone` with:
+```console
+> git submodule update --init --recursive
+```
+Similarly, when doing `pull` or `checkout` it can be handy to operate on all submodules of the repo:
 ```console
 > git pull --recurse-submodules
 > git checkout --recurse-submodules
 ```
-or for automatic support plus recommended pull ff-only global option
+
+## Build and Deploy
+Currently the code is written to support STM32 micros and therefore the build-chain utilizes `gcc`, since it is supported by STMCubeMX. On successful installation of the compiler it should appear on your path and will have a prefix of `arm-none-eabi`, so for example `arm-none-eabi-gcc`.
```

Comment:
This is out of date. The make script now points to ./motorlib/gcc/bin by default. You could make it point to a global installation with make GCC_PATH= 

Also maybe I want to remove references to cubemx at this point

Reply:
Sounds good. I updated the wording and shortened things. If folks follow the installation directions, they don't really need to hear it again anyway.